src/main/java/seedu/address/model/expense/Date.java

```diff
@@ -61,6 +61,33 @@ public static boolean isValidDate(String test) {
         return false;
     }
 
+    /**
+     * return true if {@code this} is earlier than {@param other}
+     * */
+    //@@author Jiang Chen
+    public boolean isEalierThan(Date other) {
+        if (this.fullDate.get(Calendar.YEAR) < other.fullDate.get(Calendar.YEAR)) {
```

Comment:
Extract values such as "this.fullDate.get(Calendar.YEAR)" into variables for readability